lib/io.js

```diff
@@ -51,22 +51,31 @@ BinaryDecoder.prototype = {
         var oldOffset = this._input.offset;
         var b = this.readByte();
         if (b instanceof AvroErrors.BlockDelayReadError) return b;
+        b = Long.fromBits(b, 0);
 
-        var n = b & 0x7F;
+        var L_0x7f = Long.fromNumber(0x7F);
+        var L_0xFF = Long.fromNumber(0xFF);
+
+        var n = b.and(L_0x7f);
         var shift = 7;
+        var numShifts = 0;
 
-        while ((b & 0x80) != 0) {
+        while (b.greaterThan(L_0x7f) && numShifts < 7) {
             b = this.readByte();
             if (b instanceof AvroErrors.BlockDelayReadError) {
                 //console.error("offset now is %d, old was %d, shift is %d", this._input.offset, oldOffset, shift);
                 this._input.skip(oldOffset - this._input.offset);
                 return b;
             }
-            n |= (b & 0x7F) << shift
```

Comment:
Nope, won't work.  Remember that bit shift operations convert the numbers provided to 32-bit unsigned integers.  This would fail when you try to do a shift that is greater than 30.